src/Provider/OpenRouteService/OpenRouteService.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\OpenRouteService;
+
+use Geocoder\Collection;
+use Geocoder\Exception\InvalidCredentials;
+use Geocoder\Query\GeocodeQuery;
+use Geocoder\Query\ReverseQuery;
+use Geocoder\Provider\Pelias\Pelias;
+use Geocoder\Provider\Provider;
+use Http\Client\HttpClient;
+
+final class OpenRouteService extends Pelias implements Provider
+{
+    const API_URL = 'https://api.openrouteservice.org/geocode/';
+
+    const API_VERSION = 1;
+
+    /**
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @param HttpClient $client an HTTP adapter
+     * @param string     $apiKey an API key
+     */
+    public function __construct(HttpClient $client, string $apiKey)
+    {
+        if (empty($apiKey)) {
+            throw new InvalidCredentials('No API key provided.');
+        }
+
+        $this->apiKey = $apiKey;
+        parent::__construct($client, self::API_URL, self::API_VERSION);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function geocodeQuery(GeocodeQuery $query): Collection
+    {
+        $url = $this->getGeocodeQueryUrl($query, [
+            'api_key' => $this->apiKey,
+        ]);
+
+        $url = str_replace('/v1/', '/', $url);
```

Comment:
I'm curious as to the reason this is here. I'm guessing your implementation of pelias doesn't have versioning?

Maybe better to make a change to the base class to allow a null version instead of this (which is a bit hacky)

Reply:
According to the documentation, the current version (`v2`) has `/v2/` in the url (version 1 doesn't)  !
Please switch your code to current version (`v2`) !

Reply:
Here's some clarification from the linked drupal PR:

> ORS is quite as Pelias but with some variations : API key required, and no version number in API calls. I found no easy way to handle this variations in a Drupal patch, as I cannot implement a subclass and don't want to duplicate code (I could try to use composer exclude-from-classmap to prevent autoloading, but it's weird and not guaranteed to work well).
Actually, the right way to work around the dependency problem is to provide the provider outside from this Drupal module, by contributing directly to geocoder-php, which I did: https://github.com/geocoder-php/Geocoder/pull/1026

> The good side is that anyone not using Drupal will still be able to use my code to interact with OpenRouteService. The bad side is that this feature can't be properly completed before the new provider is reviewed and included in geocoder-php. Then it will be very easy to implement the Drupal adapter.

So it sounds like it actually doesn't have a version, so maybe worth updating the base class to make it optional?

Reply:
Looking at the docs here:
https://openrouteservice.org/dev/#/api-docs/geocode/search/get

It seems to not have a version, while most of the other routes do?

Reply:
Oh sorry, you're right. I looked at v2 documentation but didn't notice the geocoding requests are without a ´/v2´ 🤦‍♂ 

I don't know how to handle this the best way.
Pelias does provide a version number ! So, I don't think it's the best way to allow to not setup a version number in Pelias "provider" !

We first need to know which version of Pelias they use in OpenRouteService (and know if they're planning to change/upgrade it in the future).

Reply:
Hi, I asked ORS about the version number: https://ask.openrouteservice.org/t/pelias-version-in-api-url/1021?u=gaelg
And yes, version number is part of the Pelias standard, that's why I made such a hack in the subclass to remove it.
As v1 in the only version available for now, do we need to wait for their response or could we just get this in, and then update the code if needed when/if we get an answer?

Reply:
I'd prefer to update the base pelias class to make the `$version` nullable.

I think it's a lot less hacky than making the change here.

https://github.com/geocoder-php/Geocoder/blob/master/src/Provider/Pelias/Pelias.php#L44-L49

Something like this 
```php
public function __construct(HttpClient $client, string $root, ?int $version = 1)
    {
		$this->root = $version !== null
             ? sprintf('%s/v%d', rtrim($root, '/'), $version);
             : $root


        $this->version = $version;
        parent::__construct($client);
    }

Reply:
Haha, I'd prefer the opposite ; not to touch the Pelias provider because the version number is part of Pelias !

@Nyholm What do you think ?

Reply:
Fair enough haha.

Happy to defer to @Nyholm

Reply:
Oh, interesting. 
@jbelien says that version should be a part of the url according to Pelias specification. But the OpenRouteService has decided not to do that. That is why we have to write this little hack with `str_replace`, correct?

Are there any other Pelias providers that need this hack? Eg: Is OpenRouteService a super special thing or does this happens a lot? If it happens a lot I would consider changing our abstraction (`Pelias` provider). 

If this is our only current case, I would prefer if we kept this hack here, or preferably, add/update the `OpenRouteService` to avoid the `str_replace`. 

Does it make sense?

Reply:
> That is why we have to write this little hack with str_replace, correct?

Yes!

> Are there any other Pelias providers that need this hack?

AFAIK, no other Pelias provider do that!

> If this is our only current case, I would prefer if we kept this hack here, or preferably, add/update the OpenRouteService to avoid the str_replace.

I didn't get that part. I would prefer to keep the special case locally to OpenRouteService provider (and not touch the Pelias provider).
But the `str_replace` is proposed in this OpenRouteService PR.
Should we find a "better" hack than `str_replace` ? Is it what you mean ?


Thanks :)

Reply:
Oh, Im missing a word.. =)

> "add/update the OpenRouteService's constructor to avoid the str_replace."

Ie write a new value to `$this->root`

Reply:
Yes! That would work indeed! 🎉 👍

Reply:
Like this? https://github.com/gaelg/Geocoder/commit/1118bbb8256f15e2375ca58133e99a8c59b4a020